codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -322,17 +359,12 @@ void returnProcessedBytes(int delta) throws Http2Exception {
         }
 
         void consumeBytes(ChannelHandlerContext ctx, int numBytes) throws Http2Exception {
-            if (stream.id() == CONNECTION_STREAM_ID) {
-                throw new UnsupportedOperationException("Returning bytes for the connection window is not supported");
-            }
-            if (numBytes <= 0) {
-                throw new IllegalArgumentException("numBytes must be positive");
-            }
-
             // Return bytes to the connection window
-            FlowState connectionState = connectionState();
-            connectionState.returnProcessedBytes(numBytes);
-            connectionState.writeWindowUpdateIfNeeded(ctx);
+            if (stream.id() != CONNECTION_STREAM_ID) {
```

Comment:
Agreed this is a little ugly.  I got rid of the special connection logic and changed the code to call `consumeBytes` on both states separately.